system/jlib/jptree.cpp

```diff
@@ -73,20 +73,59 @@ IPropertyTreeIterator *createNullPTreeIterator() { return LINK(nullPTreeIterator
 
 //===================================================================
 
+#ifdef USE_READONLY_ATOMTABLE
+RONameTable *AttrStrUnionWithTable::roNameTable = nullptr;
+#endif
 static AtomRefTable *keyTable = nullptr;
 static AtomRefTable *keyTableNC = nullptr;
 static CAttrValHashTable *attrHT = nullptr;
 
+
 #ifdef USE_FREEALLOCATOR
 static CriticalSection hashcrit;
 static AttrValue **freelist = nullptr;
 static unsigned freelistmax = 0;
 static CLargeMemoryAllocator freeallocator((memsize_t)-1, 0x1000*sizeof(AttrValue), true);
 #endif
 
+#ifdef USE_READONLY_ATOMTABLE
+static const char * roAttributes[] =
+{
+#include "jptree-attrs.hpp"    // potentially auto-generated
```

Comment:
should this really be an .hpp ? As it's not a strictly valid code on it's own.
Perhaps better if the whole definition was in header.